From c4323f6c23d4cd1b8bdc8762852e6d6b95eb4904 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Thu, 18 Jul 2019 13:49:15 +0800
Subject: [PATCH 616/741] felix: export felix_ptp_gettime()

commit ac3de3fa83aa567e2ae64e0550f086a2e763fff1 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Export felix_ptp_gettime() for global usage. The Felix switch
device driver and TSN driver may need this function.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/mscc/felix_ptp.c | 3 ++-
 drivers/net/ethernet/mscc/ocelot.h    | 3 +++
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mscc/felix_ptp.c b/drivers/net/ethernet/mscc/felix_ptp.c
index cfa972d412e3..a11d89607491 100644
--- a/drivers/net/ethernet/mscc/felix_ptp.c
+++ b/drivers/net/ethernet/mscc/felix_ptp.c
@@ -6,7 +6,7 @@
 
 #include "ocelot.h"
 
-static int felix_ptp_gettime(struct ptp_clock_info *ptp, struct timespec64 *ts)
+int felix_ptp_gettime(struct ptp_clock_info *ptp, struct timespec64 *ts)
 {
 	struct ocelot *ocelot = container_of(ptp, struct ocelot, ptp_caps);
 	u32 val, ns, sec_lsb, sec_msb;
@@ -34,6 +34,7 @@ static int felix_ptp_gettime(struct ptp_clock_info *ptp, struct timespec64 *ts)
 
 	return 0;
 }
+EXPORT_SYMBOL_GPL(felix_ptp_gettime);
 
 static int felix_ptp_settime(struct ptp_clock_info *ptp,
 			     const struct timespec64 *ts)
diff --git a/drivers/net/ethernet/mscc/ocelot.h b/drivers/net/ethernet/mscc/ocelot.h
index 9f28990be97e..6fcb110a20d4 100644
--- a/drivers/net/ethernet/mscc/ocelot.h
+++ b/drivers/net/ethernet/mscc/ocelot.h
@@ -545,9 +545,12 @@ int ocelot_probe_port(struct ocelot *ocelot, u8 port,
 #ifdef CONFIG_MSCC_FELIX_SWITCH_PTP_CLOCK
 int felix_ptp_init(struct ocelot *ocelot);
 void felix_ptp_remove(struct ocelot *ocelot);
+int felix_ptp_gettime(struct ptp_clock_info *ptp, struct timespec64 *ts);
 #else
 static inline int felix_ptp_init(struct ocelot *ocelot) { return 0; }
 static inline void felix_ptp_remove(struct ocelot *ocelot) { }
+static inline int felix_ptp_gettime(struct ptp_clock_info *ptp,
+				    struct timespec64 *ts) { return 0; }
 #endif
 
 extern struct notifier_block ocelot_netdevice_nb;
-- 
2.17.1

